main.go

```diff
@@ -41,50 +51,60 @@ func main() {
 }
 
 func run(ctx context.Context, args []string) error {
-	flags := flag.NewFlagSet("", flag.ExitOnError)
-
 	var (
-		debug bool
-
-		owmAPIKey string
-		owmQuery  string
-
-		githubAPIToken string
+		debug      bool
+		configPath string
 	)
-	flags.BoolVar(&debug, "debug", false, "Enable debug logging")
-	flags.StringVar(&owmAPIKey, "owm.api-key", "", "OpenWeather API key")
-	flags.StringVar(&owmQuery, "owm.query", "Berlin,de", "OpenWeather API query, city name, state and country code divided by comma")
 
-	flags.StringVar(&githubAPIToken, "github.token", "", "GitHub API token")
+	flags := flag.NewFlagSet("", flag.ExitOnError)
+	flags.BoolVar(&debug, "debug", false, "Enable debug logging")
+	flags.StringVar(&configPath, "configuration", "config.yaml", "Path to configuration file")
 
 	if err := flags.Parse(args); err != nil {
 		return err
 	}
 
-	if owmAPIKey == "" || githubAPIToken == "" {
-		return fmt.Errorf("no API credentials passed: OpenWeather %q, GitHub %q", owmAPIKey, githubAPIToken)
+	f, err := os.Open(configPath)
+	if err != nil {
+		return fmt.Errorf("error opening configuration file: %v", err)
+	}
+	defer f.Close()
+
+	var cfg Config
+	dec := yaml.NewDecoder(f)
+	err = dec.Decode(&cfg)
+	if err != nil {
+		return fmt.Errorf("error parsing configuration file: %v", err)
+	}
+
+	if cfg.Owm.ApiKey == "" || cfg.Github.Token == "" {
+		return fmt.Errorf("no API credentials passed: OpenWeather %q, GitHub %q", cfg.Owm.ApiKey, cfg.Github.Token)
```

Comment:
I believe in #4 you also wanted to fix this particular line, so the CLI didn't print the secrets, right?

Reply:
Yeah, totally forgot about this. Mainly I was talking about the exchange from CLI flags to a config file but I could bake a masked PW feature into this PR.